diff --git a/node_modules/@nativescript-community/sqlite/platforms/android/include.gradle b/node_modules/@nativescript-community/sqlite/platforms/android/include.gradle
index 612a631..be968b2 100644
--- a/node_modules/@nativescript-community/sqlite/platforms/android/include.gradle
+++ b/node_modules/@nativescript-community/sqlite/platforms/android/include.gradle
@@ -5,7 +5,6 @@ android {
 }
 
 dependencies {
-    // Describe plugin native Android dependencies like
-	// implementation "groupName:pluginName:ver" 
-    // EXAMPLE: implementation "com.facebook.fresco:fresco:0.9.0+"
+  implementation 'net.zetetic:sqlcipher-android:4.9.0@aar'
+  implementation 'androidx.sqlite:sqlite:2.2.0'
 }
\ No newline at end of file
diff --git a/node_modules/@nativescript-community/sqlite/platforms/android/sqlite.aar b/node_modules/@nativescript-community/sqlite/platforms/android/sqlite.aar
new file mode 100644
index 0000000..4c19864
Binary files /dev/null and b/node_modules/@nativescript-community/sqlite/platforms/android/sqlite.aar differ
diff --git a/node_modules/@nativescript-community/sqlite/platforms/ios/Podfile b/node_modules/@nativescript-community/sqlite/platforms/ios/Podfile
index 09f9b5e..c94dbf3 100644
--- a/node_modules/@nativescript-community/sqlite/platforms/ios/Podfile
+++ b/node_modules/@nativescript-community/sqlite/platforms/ios/Podfile
@@ -1 +1 @@
-pod 'FMDB'
\ No newline at end of file
+pod 'FMDB/SQLCipher'
\ No newline at end of file
diff --git a/node_modules/@nativescript-community/sqlite/sqlite.android.d.ts b/node_modules/@nativescript-community/sqlite/sqlite.android.d.ts
index e93d850..1ae1c47 100644
--- a/node_modules/@nativescript-community/sqlite/sqlite.android.d.ts
+++ b/node_modules/@nativescript-community/sqlite/sqlite.android.d.ts
@@ -3,14 +3,15 @@ import { SQLiteDatabaseBase } from './sqlitedatabase.android';
 export declare class SQLiteDatabase extends SQLiteDatabaseBase implements ISQLiteDatabase {
     open(): boolean;
 }
-export declare function wrapDb(db: android.database.sqlite.SQLiteDatabase, options?: {
+export declare function wrapDb(db: net.zetetic.database.sqlcipher.SQLiteDatabase, options?: {
     readOnly?: boolean;
     transformBlobs?: boolean;
     threading?: boolean;
 }): SQLiteDatabase;
-export declare const openOrCreate: (filePath: string, options?: {
+export declare const openOrCreate: (filePath: string,  options?: {
     threading?: boolean;
     transformBlobs?: boolean;
     flags?: number;
+    password?: string;
 }) => SQLiteDatabase;
 export declare const deleteDatabase: (filePath: string) => boolean;
diff --git a/node_modules/@nativescript-community/sqlite/sqlite.android.js b/node_modules/@nativescript-community/sqlite/sqlite.android.js
index 9aa9075..dc2190c 100644
--- a/node_modules/@nativescript-community/sqlite/sqlite.android.js
+++ b/node_modules/@nativescript-community/sqlite/sqlite.android.js
@@ -1,22 +1,32 @@
-import { Application } from '@nativescript/core';
 import { SQLiteDatabaseBase } from './sqlitedatabase.android';
-function createDb(dbName, flags) {
+
+function createDb(dbName, password, flags) {
     if (dbName === ':memory:') {
         //noinspection JSUnresolvedVariable
-        return android.database.sqlite.SQLiteDatabase.create(flags);
+        return net.zetetic.database.sqlcipher.SQLiteDatabase.openOrCreateDatabase(
+            ":memory:",
+            password,
+            null,
+            null
+        );
     }
     if (dbName.indexOf('/') >= 0) {
-        return android.database.sqlite.SQLiteDatabase.openDatabase(dbName, null, flags !== undefined ? flags : android.database.sqlite.SQLiteDatabase.CREATE_IF_NECESSARY | android.database.sqlite.SQLiteDatabase.NO_LOCALIZED_COLLATORS);
+        return net.zetetic.database.sqlcipher.SQLiteDatabase.openDatabase(dbName, password, null, flags !== undefined ? flags : net.zetetic.database.sqlcipher.SQLiteDatabase.CREATE_IF_NECESSARY | net.zetetic.database.sqlcipher.SQLiteDatabase.NO_LOCALIZED_COLLATORS, null, null);
     }
     else {
-        const activity = Application.android.foregroundActivity || Application.android.startActivity;
-        return activity.openOrCreateDatabase(dbName, flags !== undefined ? flags : android.app.Activity.MODE_PRIVATE, null);
+        return net.zetetic.database.sqlcipher.SQLiteDatabase.openOrCreateDatabase(
+            dbName,
+            password,
+            null,
+            null,
+            null,
+        );
     }
 }
 export class SQLiteDatabase extends SQLiteDatabaseBase {
     open() {
         if (!this.db) {
-            this.db = createDb(this.filePath, this.flags);
+            this.db = createDb(this.filePath, this.password, this.flags);
             if (this.threading && !this.worker) {
                 this.worker = new Worker('./worker');
                 this.worker.onmessage = this.onWorkerMessage;
@@ -35,5 +45,5 @@ export const openOrCreate = (filePath, options) => {
     obj.open();
     return obj;
 };
-export const deleteDatabase = (filePath) => android.database.sqlite.SQLiteDatabase.deleteDatabase(new java.io.File(filePath));
+export const deleteDatabase = (filePath) => net.zetetic.database.sqlcipher.SQLiteDatabase.deleteDatabase(new java.io.File(filePath));
 //# sourceMappingURL=sqlite.android.js.map
\ No newline at end of file
diff --git a/node_modules/@nativescript-community/sqlite/sqlite.ios.d.ts b/node_modules/@nativescript-community/sqlite/sqlite.ios.d.ts
index d4cb10e..720b2ae 100644
--- a/node_modules/@nativescript-community/sqlite/sqlite.ios.d.ts
+++ b/node_modules/@nativescript-community/sqlite/sqlite.ios.d.ts
@@ -2,11 +2,13 @@ import { SQLiteDatabase as ISQLiteDatabase, SqliteParams, SqliteRow } from './sq
 export declare class SQLiteDatabase implements ISQLiteDatabase {
     db: FMDatabase;
     transformBlobs: boolean;
+    password: string | null;
     filePath: string;
     constructor(filePathOrDb: string | FMDatabase, options?: {
         threading?: boolean;
         transformBlobs?: boolean;
         readOnly?: boolean;
+        password?: string;
     });
     isOpen: boolean;
     open(): boolean;
@@ -26,10 +28,12 @@ export declare function openOrCreate(filePath: string, flags?: number, options?:
     readOnly?: boolean;
     transformBlobs?: boolean;
     threading?: boolean;
+    password?: string;
 }): SQLiteDatabase;
 export declare function wrapDb(db: FMDatabase, options?: {
     readOnly?: boolean;
     transformBlobs?: boolean;
     threading?: boolean;
+    password?: string;
 }): SQLiteDatabase;
 export declare const deleteDatabase: (filePath: string) => boolean;
diff --git a/node_modules/@nativescript-community/sqlite/sqlite.ios.js b/node_modules/@nativescript-community/sqlite/sqlite.ios.js
index 3af9e7a..3a65a1f 100644
--- a/node_modules/@nativescript-community/sqlite/sqlite.ios.js
+++ b/node_modules/@nativescript-community/sqlite/sqlite.ios.js
@@ -384,6 +384,7 @@ export class SQLiteDatabase {
             this.filePath = filePathOrDb;
         }
         this.transformBlobs = !options || options.transformBlobs !== false;
+        this.password = options && options.password ? options.password : null;
     }
     open() {
         if (!this.db) {
@@ -393,6 +394,10 @@ export class SQLiteDatabase {
         if (!this.isOpen) {
             this.isOpen = this.db.open();
         }
+        if(this.isOpen && this.password) {
+            this.db.setKey(this.password);
+        }
+
         return this.isOpen;
     }
     close() {
diff --git a/node_modules/@nativescript-community/sqlite/sqlitedatabase.android.d.ts b/node_modules/@nativescript-community/sqlite/sqlitedatabase.android.d.ts
index 80bec73..5cc6438 100644
--- a/node_modules/@nativescript-community/sqlite/sqlitedatabase.android.d.ts
+++ b/node_modules/@nativescript-community/sqlite/sqlitedatabase.android.d.ts
@@ -2,14 +2,16 @@ import { SqliteParams } from './sqlite.common';
 export declare function byteArrayToBuffer(value: any): Uint8Array;
 export declare class SQLiteDatabaseBase {
     filePath: string;
-    db: android.database.sqlite.SQLiteDatabase;
+    db: net.zetetic.database.sqlcipher.SQLiteDatabase;
+    password: string | null;
     flags: any;
     transformBlobs: boolean;
-    constructor(filePathOrDb: string | android.database.sqlite.SQLiteDatabase, options?: {
+    constructor(filePathOrDb: string | net.zetetic.database.sqlcipher.SQLiteDatabase, options?: {
         threading?: boolean;
         readOnly?: boolean;
         flags?: number;
         transformBlobs?: boolean;
+        password?: string;
     });
     _isInTransaction: boolean;
     threading: boolean;
diff --git a/node_modules/@nativescript-community/sqlite/sqlitedatabase.android.js b/node_modules/@nativescript-community/sqlite/sqlitedatabase.android.js
index fdd25c9..89b91a1 100644
--- a/node_modules/@nativescript-community/sqlite/sqlitedatabase.android.js
+++ b/node_modules/@nativescript-community/sqlite/sqlitedatabase.android.js
@@ -1,4 +1,7 @@
 import { paramsToStringArray, throwError } from './sqlite.common';
+
+java.lang.System.loadLibrary('sqlcipher');
+
 export function byteArrayToBuffer(value) {
     if (!value) {
         return null;
@@ -128,7 +131,7 @@ export class SQLiteDatabaseBase {
     constructor(filePathOrDb, options) {
         this._isInTransaction = false;
         this.threading = false;
-        if (filePathOrDb instanceof android.database.sqlite.SQLiteDatabase) {
+        if (filePathOrDb instanceof net.zetetic.database.sqlcipher.SQLiteDatabase) {
             this.db = filePathOrDb;
             this.filePath = filePathOrDb.getPath();
         }
@@ -137,6 +140,7 @@ export class SQLiteDatabaseBase {
         }
         this.threading = options && options.threading === true;
         this.flags = options?.flags;
+        this.password = options?.password || null;
         this.transformBlobs = !options || options.transformBlobs !== false;
     }
     onWorkerMessage(event) {
